sov-enshrine-esdt-safe/src/deposit.rs

```diff
@@ -0,0 +1,82 @@
+use cross_chain::MAX_TRANSFERS_PER_TX;
+use multiversx_sc::imports::*;
+use operation::{aliases::OptionalValueTransferDataTuple, OperationData, TransferData};
+
+#[multiversx_sc::module]
+pub trait DepositModule:
+    multiversx_sc_modules::pause::PauseModule
+    + utils::UtilsModule
+    + cross_chain::deposit_common::DepositCommonModule
+    + cross_chain::execute_common::ExecuteCommonModule
+    + cross_chain::storage::CrossChainStorage
+    + cross_chain::events::EventsModule
+    + max_bridged_amount_module::MaxBridgedAmountModule
+{
+    #[payable("*")]
+    #[endpoint]
+    fn deposit(
+        &self,
+        to: ManagedAddress,
+        optional_transfer_data: OptionalValueTransferDataTuple<Self::Api>,
+    ) {
+        require!(self.not_paused(), "Cannot create transaction while paused");
+
+        let (fees_payment, payments) = self.check_and_extract_fee().into_tuple();
+        require!(!payments.is_empty(), "Nothing to transfer");
+        require!(payments.len() <= MAX_TRANSFERS_PER_TX, "Too many tokens");
```

Comment:
I don't see this in `sov-esdt-safe`. Do you need this in `sov-enshrine-esdt-safe`?